frontend/packages/ux-editor/src/containers/FormItemContext.tsx

```diff
@@ -82,42 +100,69 @@ export const FormItemContextProvider = ({
   }, [formItemId, formItem]);
 
   const handleContainerSave = useCallback(
-    async (id: string, updatedContainer: FormContainer): Promise<void> => {
-      await updateFormContainer({
-        id,
-        updatedContainer,
-      });
-      if (id !== updatedContainer.id) {
+    async (
+      id: string,
+      updatedContainer: FormContainer,
+      mutateOptions?: UpdateFormMutateOptions,
+    ): Promise<void> => {
+      const hasNewId = id !== updatedContainer.id;
+      await updateFormContainer(
+        {
+          id,
+          updatedContainer,
+        },
+        {
+          onSuccess: async () => {
```

Comment:
It seems like this code, or similar code is used several places in this file. Can it be refactored to a smaller, reusable component? 🤔

Reply:
Done